codebase/superdarn/src.bin/tk/tool/fit_median_filter.1.0/fit_median_filter.c

```diff
@@ -0,0 +1,322 @@
+/* fit_median_filter.c
+   ==========
+   
+Removes salt & pepper noise from a fitacf file using median filtering. 
+Filtering is performed separately for each beam and channel. 
+Output is a fitacf file.
+
+(C) Copyright 2021 E.C.Bland
+author: E.C.Bland
+
+This file is part of the Radar Software Toolkit (RST).
+
+RST is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <https://www.gnu.org/licenses/>.
+       
+Modifications:
+
+   
+*/
+
+
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <sys/stat.h>
+#include <string.h>
+#include <time.h>
+#include <zlib.h>
+
+#include "rtypes.h"
+#include "rtime.h"
+#include "option.h"
+#include "dmap.h"
+#include "rprm.h"
+#include "fitdata.h"
+
+#include "fitread.h"
+#include "fitwrite.h"
+
+#include "errstr.h"
+#include "hlpstr.h"
+
```

Comment:
Add inline comment what each number means and why if relevant

Reply:
ok, see below